src/lib/ocrFormats.js

```diff
@@ -30,34 +25,40 @@ function parseHocrAttribs(titleAttrib) {
 /** Parse an hOCR node */
 function parseHocrNode(node, endOfLine = false, scaleFactor = 1) {
   const [ulx, uly, lrx, lry] = parseHocrAttribs(node.title).bbox.map((dim) => dim * scaleFactor);
-  let width = lrx - ulx;
-  const height = lry - uly;
-  let text = node.textContent;
-  if (node.nextSibling instanceof Text) {
-    const charWidth = (lrx - ulx) / text.length;
-    const extraText = node.nextSibling.wholeText.replace(/\s+/, ' ');
-    text += extraText;
-    // Increase the width of the node to compensate for the extra characters
-    if (!endOfLine) {
-      width += charWidth * extraText.length;
-    } else if (text.slice(-1) !== '\u00AD') {
-      // Add newline if the line does not end on a hyphenation
-      text = `${text.trim()}\n`;
-    }
-  }
-
   let style = node.getAttribute('style');
   if (style) {
     style = style.replace(/font-size:.+;/, '');
   }
-  return {
-    height,
+  const spans = [{
+    height: lry - uly,
     style,
-    text,
-    width,
+    text: node.textContent,
+    width: lrx - ulx,
     x: ulx,
     y: uly,
-  };
+    isExtra: false,
+  }];
+
+  // Add an extra space span if the following text node contains something
+  if (node.nextSibling instanceof Text) {
+    let extraText = node.nextSibling.wholeText.replace(/\s+/, ' ');
+    if (endOfLine && spans[0].text.slice(-1) !== '\u00AD') {
+      // Add newline if the line does not end on a hyphenation
```

Comment:
Nit: A little bit clearer:
```suggestion
      // Add newline if the line does not end on a hyphenation (a soft hyphen)
```